pkg/services/libraryelements/api.go

```diff
@@ -56,7 +72,25 @@ func (l *LibraryElementService) createHandler(c *contextmodel.ReqContext) respon
 			cmd.FolderID = folder.ID
 		}
 	}
-	element, err := l.CreateElement(c.Req.Context(), c.SignedInUser, cmd)
+
+	if l.features.IsEnabled(featuremgmt.FlagLibraryPanelRBAC) {
+		allowed, err := l.AccessControl.Evaluate(
+			c.Req.Context(),
+			c.SignedInUser,
+			ac.EvalPermission(
+				dashboards.ActionFoldersWrite,
+				dashboards.ScopeFoldersRoot+":id:"+fmt.Sprint(cmd.FolderID),
+			),
+		)
+		if err != nil {
+			return toLibraryElementError(err, "Failed to evaluate permissions")
+		}
+		if !allowed {
+			return response.Error(http.StatusForbidden, "You do not have the permissions needed to create a library element in this folder", nil)
+		}
+	}
```

Comment:
We still evaluate access to the folder from [here](https://github.com/grafana/grafana/blob/38c3483594080b6d7e12d94f95bcacf67c3ac368/pkg/services/libraryelements/database.go#L165), right? I think that is probably enough, and this additional check is not needed.